src/utils/daos.c

```diff
@@ -598,6 +598,8 @@ args_free(struct cmd_args_s *ap)
 	D_FREE(ap->user);
 	D_FREE(ap->group);
 	D_FREE(ap->principal);
+	D_FREE(ap->pool_label);
+	D_FREE(ap->cont_label);
```

Comment:
There is a similar cleanup function in https://github.com/daos-stack/daos/blob/master/src/control/cmd/daos/util.go#L184

Because these were copied using D_STRNDUP(), you'll want to use C.free_daos_alloc() there to make sure that NLT is happy. I'm kind of surprised that NLT is not complaining about this, actually... Maybe I'm missing where these are freed in the new tool.

Reply:
so this is the daos_old file. are you referring to the strndups made in: src/control/cmd/daos/util.c ??
would something like this be needed:

```
diff --git a/src/control/cmd/daos/util.go b/src/control/cmd/daos/util.go
index c8291df..21e268a 100644
--- a/src/control/cmd/daos/util.go
+++ b/src/control/cmd/daos/util.go
@@ -188,6 +188,8 @@ func freeCmdArgs(ap *C.struct_cmd_args_s) {
 
        freeString(ap.sysname)
        C.free_daos_alloc(unsafe.Pointer(ap.dfs_path))
+       C.free_daos_alloc(unsafe.Pointer(ap.pool_label))
+       C.free_daos_alloc(unsafe.Pointer(ap.cont_label))
 
        if ap.props != nil {
                C.daos_prop_free(ap.props)
```

Reply:
should be fixed now

Reply:
Looks good, thanks.